fluent-bundle/src/types.rs

```diff
@@ -111,14 +112,69 @@ impl<'source> From<&ast::InlineExpression<'source>> for DisplayableNode<'source>
     }
 }
 
-#[derive(Debug, PartialEq, Clone)]
+pub trait FluentType: fmt::Debug + fmt::Display + AnyEq + 'static {
+    fn duplicate(&self) -> Box<dyn FluentType>;
```

Comment:
Fwiw you can do the same thing here with an AnyClone trait

Reply:
How would a signature of that trait look like? `fn clone(&self) -> Box<dyn Any>;`? That feels off...

Reply:
Hmm, fair. Yeah, this is trickier.